synapse/config/server.py

```diff
@@ -928,6 +929,7 @@ def parse_listener_def(num: int, listener: Any) -> ListenerConfig:
             resources=resources,
             additional_resources=listener.get("additional_resources", {}),
             tag=listener.get("tag"),
+            request_id_header=listener.get("request_id_header"),
```

Comment:
Is there a better spot in the config for this? I put it next to `x_forwarded` because it's also another header thing

```yaml
listeners:
  - port: 8008
    tls: false
    type: http
    x_forwarded: true
    request_id_header: "cf-ray"
    bind_addresses: ['::1', '127.0.0.1', '0.0.0.0']
```

Reply:
I think this spot makes sense, next to the `x_forwarded`, as you say.

Plus it makes sense to want to control this per listener, eg replication likely doesn't go through a reverse proxy.